### DISCUSSION TITLE
Kapı Kilidini Aç

### DISCUSSION THREAD

**Discussion by tolga0102** [sonsuzus/sonsuzus#380] [Soru Cevap]
**Sorun**

Arkadaşımız Monk sonunda **Programlama Sırları Tapınağı** buldu. Ancak tapınağın kapısı sıkıca kilitlenmiştir. Şimdi, tapınağın kurallarına göre, Monk'un kapının kilidini açmak için özel bir dilde bir **Gizli Şifre** girmesi gerekiyor. Bu dil, İngilizce'den farklı olarak K alfabesinden oluşur. Bu gizli parolanın özellikleri şunlardır:

  1. N karakter uzunluğundadır.

  2. Özel dile ait sadece K karakterlerinden oluşur.

  3. Gizli kodda özel dile ait her karakter **en fazla bir kez** kullanılmıştır.

Şimdi, Monk'un ideal parolanın ne olabileceği konusunda hiçbir fikri yok ve sizin yardımınıza ihtiyacı var. Monk'a **Modulo** 10^9 + 7 için toplam farklı aday Dizgi sayısını bulmasında yardım etmeniz gerekiyor.

**Giriş Formatı:**

İlk satır, test durumlarının sayısını gösteren tek bir T tamsayısını içerir. Sonraki T satırlarının her biri, sırasıyla Gizli Parolanın uzunluğunu ve kullanılacak Özel dilin karakter sayısını gösteren iki tam sayı N ve K içerir.

**Çıkış Formatı:**

Her bir test durumu için, olası farklı gizli şifrelerin sayısını çıkarın **Modulo** 10^9 + 7.

**Kısıtlamalar:**

1 < T < 10
1 < N < K < 10^5

**Not:**

Her bir öğenin ağırlıklarını değil, değerini yazdırmanız gerekir.

```
**Örnek Giriş**
1
3 3

**Örnek Çıktı**
6
```

**Comment by tolga0102** [comment]
https://www.hackerearth.com/practice/math/number-theory/basic-number-theory-1/practice-problems/algorithm/name-count/

**Comment by EfeMertIrez** [comment]
```
mod = 1000000007

def compute_combination(n, k):
    result = 1
    for i in range(1, k + 1):
        result = (result * (n - i + 1)) % mod
        result = (result * pow(i, mod - 2, mod)) % mod
    return result

t = int(input())
for _ in range(t):
    n, k = map(int, input().strip().split())
    print(compute_combination(n, k))
```
Bu kod, kullanıcıdan test vakalarını alır, her bir test için kombinasyon hesaplamalarını yapar ve sonuçları 10^9 + 7 (1000000007) ile mod alarak yazdırır.

**Reply by sonsuzus** [reply]
Yanlış cevap diyor.
